### PR TITLE
[Automation]Fix the issue of processing PSCustomObject

### PR DESCRIPTION
**Pull request by wyunchi-ms**
## Description

1. Fix the issue of processing `PSCustomObject`
2. Fix the issue of processing `Array`

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Review by dingmeng-xue — DISMISSED** [review]

File: src/Automation/Automation/ChangeLog.md [hunk 1]
```diff
@@ -18,6 +18,7 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Fix the issue of processing `PSCustomObject` and `Array`.
```
Comment (dingmeng-xue) [inline src/Automation/Automation/ChangeLog.md:21]: Is there github issue id? Please use past tense.

Comment (wyunchi-ms) [reply src/Automation/Automation/ChangeLog.md:21]: This is an ICM issue, should I create a github issue?

Comment (dingmeng-xue) [reply src/Automation/Automation/ChangeLog.md]: No. let's mitigate IcM ticket too.

**Review by dingmeng-xue — APPROVED** [review]

**Pull request merged**